feature_engine/encoding/mean_encoding.py

```diff
@@ -127,6 +128,7 @@ def fit(self, X: pd.DataFrame, y: pd.Series):
             The target.
         """
 
+        X, y = _check_X_y_pd_np_mismatch(X, y)
         X = self._check_fit_input_and_variables(X)
 
         if not isinstance(y, pd.Series):
```

Comment:
I would remove these lines of code, and I would make this part of check_X_y, to reduce boilerplate